app/src/main/java/com/github/se/cyrcle/model/user/UserRepositoryFirestore.kt

```diff
@@ -41,7 +50,7 @@ constructor(private val db: FirebaseFirestore, private val auth: FirebaseAuth) :
                 val userDetails = deserializeUserDetails(detailsDocument.data!!)
                 onSuccess(User(userPublic, userDetails))
               }
-              .addOnFailureListener(onFailure)
+              .addOnFailureListener { onSuccess(User(userPublic, null)) }
```

Comment:
It feels weird to have an `onSuccess` for an `FailureListener`. If it is an error, please fix, but if it is intended, it would be nice to have a small comment asserting that this is expected.

Reply:
It's very much intended. We let the database deny our request if we don't have access instead of checking in viewmodel if we have permission. 

This is more of a implementation choice. I will add a comment to explain this reasoning

Reply:
Thank you for the clarification